src/plugins/discord/utils.ts

```diff
@@ -17,6 +17,27 @@ export const truncateString = (str: string, length: number): string => {
   return str;
 };
 
+/**
+ * Sanitizes a string for Discord Rich Presence activity, ensuring it meets length requirements.
+ * @param input - The string to sanitize.
+ * @param fallback - A fallback string to use if the input is empty or whitespace. Defaults to 'undefined'.
+ * @returns The sanitized string, compliant with Discord's requirements.
+ */
+export function sanitizeActivityText(input: string | undefined, fallback: string = 'undefined'): string {
+  const text = (input && input.trim()) ? input.trim() : fallback.trim();
+  let safeString = truncateString(text, 128);
+
+  if (safeString.length === 0) {
+    return 'undefined'; // change if necessary
```

Comment:
Any reason to not use the existing fallback?

```suggestion
    return fallback;
```